Build failure: SPAdes

### Steps To Reproduce

Steps to reproduce the behavior:
1. nix build .#SPAdes

### Build log

```
/build/SPAdes-3.15.5/src/common/sequence/nucl.hpp: In function 'char nucl(char)':
/build/SPAdes-3.15.5/src/common/sequence/nucl.hpp:107:15: error: 'uint32_t' does not name a type
  107 |         const uint32_t nucl_map = 'A' + ('C' << 8) + ('G' << 16) + ('T' << 24);
      |               ^~~~~~~~
```
Log file: [SPAdes.txt](https://github.com/NixOS/nixpkgs/files/13998590/SPAdes.txt)

### Additional context

* https://github.com/NixOS/nixpkgs/blob/master/pkgs/applications/science/biology/spades/default.nix
* http://cab.spbu.ru/software/spades/

* I've opened issue to reference when marking this broken. I don't use this software.

### Notify maintainers

@bzizou

### Metadata

```console
 - system: `"x86_64-linux"`
 - host os: `Linux 6.1.72, NixOS, 24.05 (Uakari), 24.05.20240115.c3e128f`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.1`
 - channels(root): `""`
 - channels(intj): `""`
 - nixpkgs: `/etc/nix/channels/nixpkgs`
```

# PR for fixing issue (Thanks to wegank!)

https://github.com/NixOS/nixpkgs/pull/282374

--- a/pkgs/applications/science/biology/spades/default.nix
+++ b/pkgs/applications/science/biology/spades/default.nix
@@ -17,6 +17,11 @@ stdenv.mkDerivation rec {
 
   sourceRoot = "${pname}-${version}/src";
 
+  env.CXXFLAGS = toString [
+    # GCC 13: error: 'uint32_t' does not name a type
+    "-include cstdint"
+  ];
+
   meta = with lib; {
     description = "St. Petersburg genome assembler: assembly toolkit containing various assembly pipelines";
     license = licenses.gpl2Only;

